task_sdk/src/airflow/sdk/execution_time/context.py

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import structlog
+
+if TYPE_CHECKING:
+    from airflow.sdk.execution_time.comms import ConnectionResult
+
+
+def _convert_connection_result_conn(conn_result: ConnectionResult):
+    from airflow.sdk.definitions.connection import Connection
+
+    return Connection(**conn_result.model_dump(exclude={"type"}, by_alias=True))
+
+
+def get_connection(conn_id: str):
+    # TODO: This should probably be moved to a separate module like `airflow.sdk.execution_time.comms`
+    #   or `airflow.sdk.execution_time.connection`
+    #   A reason to not move it to `airflow.sdk.execution_time.comms` is that it
+    #   will make that module depend on Task SDK, which is not ideal because we intend to
+    #   keep Task SDK as a separate package than execution time mods.
+    from airflow.sdk.execution_time.comms import GetConnection
+    from airflow.sdk.execution_time.task_runner import SUPERVISOR_COMMS
+
+    log = structlog.get_logger(logger_name="task")
+    SUPERVISOR_COMMS.send_request(log=log, msg=GetConnection(conn_id=conn_id))
+    msg = SUPERVISOR_COMMS.get_message()
+    return _convert_connection_result_conn(msg)
+
+
+class ConnectionAccessor:
+    """Wrapper to access Connection entries in template."""
+
+    def __init__(self) -> None:
+        self.var: Any = None
+
+    def __getattr__(self, key: str) -> Any:
+        self.var = get_connection(key)
+        return self.var
+
+    def __repr__(self) -> str:
+        return str(self.var)
+
+    def __eq__(self, other):
+        return self.var == other
+
+    def get(self, key: str, default_conn: Any = None) -> Any:
```

Comment:
This will have to return a `Connection | None`

Reply:
It can return any default because `default_conn` can be `Any`.

https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#airflow-connections-in-templates

Example:

```py
{{ conn.get('my_conn_id', {"host": "host1", "login": "user1"}).host }}
```

Check: https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#airflow-connections-in-templates

It is also mostly a copy of existing functionality: https://github.com/apache/airflow/blob/9a6f87e02a3fbb8c8d6c10b36313d9067e8a2ca6/airflow/utils/context.py#L132-L155